diff --git a/Foundation/NSString.swift b/Foundation/NSString.swift
index 9940a913..842e443b 100644
--- a/Foundation/NSString.swift
+++ b/Foundation/NSString.swift
@@ -1649,3 +1649,22 @@ extension NSString : _StructTypeBridgeable {
         return _StructType._unconditionallyBridgeFromObjectiveC(self)
     }
 }
+
+extension NSString : CVarArg {
+    @inlinable // c-abi
+    public var _cVarArgEncoding: [Int] {
+        return _encodeBitsAsWords(unsafeBitCast(self, to: CFString.self))
+    }
+}
+
+extension String : CVarArg, _CVarArgObject {
+    @inlinable // c-abi
+    public var _cVarArgObject: CVarArg {
+        return NSString(string: self)
+    }
+
+    @inlinable // c-abi
+    public var _cVarArgEncoding: [Int] {
+        fatalError("_cVarArgEncoding must be called on NSString instead")
+    }
+}
